R/stores.R

```diff
@@ -363,16 +362,36 @@ HttpStore <- R6::R6Class("HttpStore",
     headers = NULL,
     client = NULL,
     zmetadata = NULL,
-    mem_get = NULL,
-    cache_time_seconds = 3600,
+    make_request_memoized = NULL,
+    cache_enabled = NULL,
+    cache_time_seconds = NULL,
     make_request = function(item) {
       key <- item_to_key(item)
-      
-      # mem_get caches in memory on a per-session basis.
-      res <- private$mem_get(private$client, 
-                             paste(private$base_path, key, sep="/"))
-      
-      return(res)
+
+      # For some reason, the crul::HttpClient fails in parallel settings
+      # (when used inside foreach %dopar% loops). This alternative
+      # with HttpRequest and AsyncVaried seems to work.
+      # Reference: https://docs.ropensci.org/crul/articles/async.html
```

Comment:
Is there maybe an argument to switch to a more traditional http client here? Seems like there's a lot of complexity coming together.

Maybe another option is to have one simple pathway for metadata gets and another non-caching parallelized get utility for iterating over chunks?